mlflow/gateway/config.py

```diff
@@ -61,7 +62,7 @@ def _missing_(cls, value):
         raise MlflowException.invalid_parameter_value(f"Invalid OpenAI API type '{value}'")
 
 
-class OpenAIConfig(BaseModel, extra=Extra.allow):
+class OpenAIConfig(ConfigModel):
     openai_api_key: str
     openai_api_type: OpenAIAPIType = OpenAIAPIType.OPENAI
     openai_api_base: Optional[str] = None
```

Comment:
Unrelated to this PR, do we need the `<provider_name>_` prefix?

Reply:
We've already implemented this for every provider on Databricks and included it in our docs, so we'll have to live with this for now :)

Reply:
Got it, if so, can we add `cohere_` to the cohere config?

Reply:
Yes, definitely. Same for Anthropic if we’re missing the prefix there too. We should keep this consistent with DB protos

Reply:
@harupy can you make that change?

Reply:
Ok, I'll make the change